record/record.go

```diff
@@ -24,11 +24,12 @@
 //		records := record.NewReader(r)
 //		for {
 //			rec, err := records.Next()
-//			if err == io.EOF {
+//			if err == io.EOF || err == io.ErrUnexpectedEOF {
 //				break
 //			}
 //			if err != nil {
-//				return nil, err
+//				r.Recover()  // Skip corrupted records.
```

Comment:
Makes sense. If Recover were to not return an error, you could still call Next, it would just result in the same error again, and then you'd call Recover. Having Recover return an error helps break that cycle. Addressed the sample code as requested.

The comment is over 80 chars in the line, but I it's more readable. If you want me to wrap it, I will.